### ISSUE TITLE
Support HTTP method PATCH

### ISSUE THREAD

**Issue by robotdan** [FusionAuth/fusionauth-issues#121]
### Problem
Currently the update API `PUT` requires the entire object. Anything omitted will be deleted or perhaps fail validation if it is required. 

### Solution
It would be super nice to be able to update an object with just a change using the `PATCH` HTTP method and have it merge the JSON body with the current object. 

- [x] Support the `PATCH` HTTP method
- [x] Update client libraries with `PATCH` capability and corresponding methods
- [x] Update API doc for each API that will support `PATCH`

### Alternatives/workarounds
Retrieve an object using `GET`, update what you want and then update using `PUT`. 

### Additional context
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PATCH

### How to vote
Please give us a thumbs up or thumbs down as a reaction to help us prioritize this feature. Feel free to comment if you have a particular need or comment on how this feature should work.

**Comment by robotdan** [comment]
This feature is in progress!

**Comment by robotdan** [comment]
Tech preview coming shortly..

**Comment by robotdan** [comment]
Available in 1.11.0 for a few APIs, provide feedback if you choose to use them. The PATCH is not available in the client libraries yet.

**Comment by matthew-altman** [comment]
The remaining APIs will support PATCH in 1.12.0
```
/api/application/role
/api/consent
/api/email/template
/api/group
/api/identity-provider
/api/integration
/api/lambda
/api/system-configuration
/api/tenant
/api/theme
/api/user-action
/api/user-action-reason
/api/user/consent
/api/webhook
```
